src/components/app/app.jsx

```diff
@@ -20,7 +20,9 @@ const App = () => {
         <Route exact path={RoutePaths.MAIN_PAGE}>
           <MainPage />
         </Route>
-        <PrivateRoute path={RoutePaths.FAVORITES_PAGE} exact component={FavoritesPage} />
+        <Route path ={RoutePaths.FAVORITES_PAGE} exact>
+          <PrivateRoute component={() => <FavoritesPage />} noAuth={() => <Redirect to={RoutePaths.LOGIN_PAGE}/>} />
+        </Route>
```

Comment:
как то сложно выглядит. лучше было бы сделать вот так
```suggestion
          <PrivateRoute path ={RoutePaths.FAVORITES_PAGE} exact component={() => <FavoritesPage />} noAuth={() => <Redirect to={RoutePaths.LOGIN_PAGE}/>} />
```